libraries/networking/src/udt/Socket.h

```diff
@@ -0,0 +1,68 @@
+//
+//  Socket.h
+//  libraries/networking/src/udt
+//
+//  Created by Stephen Birarda on 2015-07-20.
+//  Copyright 2015 High Fidelity, Inc.
+//
+//  Distributed under the Apache License, Version 2.0.
+//  See the accompanying file LICENSE or http://www.apache.org/licenses/LICENSE-2.0.html
+//
+
+#pragma once
+
+#ifndef hifi_Socket_h
+#define hifi_Socket_h
+
+#include <functional>
+
+#include <QtCore/QObject>
+#include <QtNetwork/QUdpSocket>
+
+#include "../HifiSockAddr.h"
+#include "Packet.h"
+
+namespace udt {
+
+using VerifyPacketOperator = std::function<bool(const Packet&)>;
+using VerifiedPacketCallback = std::function<void(std::unique_ptr<Packet>)>;
+
+class Socket : public QObject {
+    Q_OBJECT
+public:
+    Socket(QObject* object = 0);
+    
+    quint16 localPort() const { return _udpSocket.localPort(); }
+    
+    qint64 writeUnreliablePacket(const Packet& packet, const HifiSockAddr& sockAddr);
+    
+    qint64 writeDatagram(const char* data, qint64 size, const HifiSockAddr& sockAddr)
+        { return writeDatagram(QByteArray::fromRawData(data, size), sockAddr); }
+    qint64 writeDatagram(const QByteArray& datagram, const HifiSockAddr& sockAddr);
+    
+    void bind(const QHostAddress& address, quint16 port = 0) { _udpSocket.bind(address, port); }
+    void rebind();
+    
+    void setVerifyPacketOperator(VerifyPacketOperator verifyPacketOperator) { _verifyPacketOperator = verifyPacketOperator; }
+    void setVerifiedPacketCallback(VerifiedPacketCallback verifiedPacketCallback)
```

Comment:
I was a little confused when first reading these names, I wonder if adding a verb here would make it more clear. examples: `OnReceiveVerifiedPacketCallback` `HandleVerifiedPacketCallback` - it wasn't clear to me at first that this was the callback to handle incoming packets.